### PR TITLE
fix(sql): fix null returned from min() on int/long/date/ts column with max values only

### LINKED ISSUE #2414: min(int), min(long) returns wrong result
### Describe the bug

min(int) and min(long) return null if input contains only max values  and nulls . 

### To reproduce

```sql
create table testl( l long );

insert into testl select 9223372036854775807L from long_sequence(1);
select * from testl;
--returns 9223372036854775807

select min(l) from testl ;
--returns null

create table testi( i int );

insert into testi select 2147483647 from long_sequence(1);
select * from testi;
--returns 2147483647

select min(i) from testi ;
--returns null
```

### Expected Behavior

min() should return max value for int or long

### Environment

```markdown
- **QuestDB version**: 6.5.1-SNAPHSOT
- **OS**:Windows
- **Browser**:Chrome
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Fixes #2414

I checked performance of vectorized `min()` before and after on a table with 100M rows and didn't notice any difference.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 34 (88.24%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinIntVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinTimestampVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinLongVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinDateVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinLongGroupByFunction.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinIntGroupByFunction.java|1|1|100.00%|

File: core/src/main/c/share/vec_agg.cpp [hunk 1]
```diff
@@ -167,29 +167,34 @@ int64_t SUM_LONG(int64_t *pl, int64_t count) {
...
     Vec8qb bVec;
     int i;
     for (i = 0; i < count - 7; i += step) {
         _mm_prefetch(pl + i + 63 * step, _MM_HINT_T1);
         vec.load(pl + i);
-        bVec = vec == L_MIN;
-        vecMin = select(bVec, vecMin, min(vecMin, vec));
+        vecMin = select(vecMin == L_MIN, vec, vecMin);
+        vec = select(vec == L_MIN, vecMin, vec);
+        vecMin = min(vec, vecMin);
```
Comment (bluestreak01) [inline core/src/main/c/share/vec_agg.cpp:178]: How would this calculate min when initial value is the lowest possible?

Comment (puzpuzpuz) [reply core/src/main/c/share/vec_agg.cpp:178]: Both previous `select()`s make sure that we may only see `(vec == L_MIN && vecMin == L_MIN) == true` or `(vec != L_MIN && vecMin != L_MIN) == true` when we come to this line

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 34 (88.24%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinIntVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinTimestampVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinLongVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinDateVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinLongGroupByFunction.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinIntGroupByFunction.java|1|1|100.00%|

File: core/src/main/c/share/vec_agg.cpp [hunk 2]
```diff
@@ -167,29 +167,34 @@ int64_t SUM_LONG(int64_t *pl, int64_t count) {
 int64_t MIN_LONG(int64_t *pl, int64_t count) {
     Vec8q vec;
     const int step = 8;
-    Vec8q vecMin = L_MAX;
+    Vec8q vecMin = L_MIN;
```
Comment (eugenels) [inline core/src/main/c/share/vec_agg.cpp:170]: Have you considered the simpler null-count approach?
Something like:
```
min = L_MAX;
nullcount = if_add(bVec, nullcount, 1);
...
return (min == L_MAX && nullcount == count) ? L_MIN : min;
```

Comment (puzpuzpuz) [reply core/src/main/c/share/vec_agg.cpp:170]: Let me try that.

**Review by eugenels — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 34 (88.24%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinIntVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinTimestampVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinLongVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinDateVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinLongGroupByFunction.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinIntGroupByFunction.java|1|1|100.00%|

**Comment by puzpuzpuz** [comment]
This one should be ready to go.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 34 (88.24%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinIntVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinTimestampVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinLongVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinDateVectorAggregateFunction.java|7|8|87.50%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinLongGroupByFunction.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinIntGroupByFunction.java|1|1|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**